### PR TITLE
board/arm/stm32h747i_disco: add support of flash with openocd

### PR DESCRIPTION
**Pull request by ABOSTM**
Allow to flash either Cortex M4 or M7 with OpenOCD
(depending on which Board/Core has been compiled)
Command: west flash

Warning: Dependency with recent OpenOCD patch:
Windows: https://gnutoolchains.com/arm-eabi/openocd/
                 version 20200408
Linux : http://openocd.zylin.com/
           SHA1: 0a804222da63c5f849efa23b019a59e2dea76842

Edit:
OpenOCD dependancy: https://github.com/zephyrproject-rtos/openocd/issues/27

Note: It is necessary to first update ST-Link firmware (see https://github.com/zephyrproject-rtos/zephyr/pull/24671)

Known Restrictions:
---------------------
Sometimes flashing doesn't work and it is necessary to erase the flash to be able make it work again.
Debugging Cortex M7 works, but it doesn't work with Cortex M4.

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Please update doc regarding openocd compatibility, potential issues that could be faced and how to overcome.
Otherwise LGTM

**Comment by ABOSTM** [comment]
Done

**Review by erwango — APPROVED** [review]

File: boards/arm/stm32h747i_disco/doc/index.rst [hunk 1]
```diff
@@ -170,6 +180,17 @@ Use the following commands to flash either m7 or m4 target:
    $ ./STM32_Programmer_CLI -c port=SWD mode=UR -w <path_to_m7_binary>  0x8000000
    $ ./STM32_Programmer_CLI -c port=SWD mode=UR -w <path_to_m4_binary>  0x8100000
 
+Alternatively it is possible to flash with OpenOcd but with some restrictions:
+Sometimes, flashing is not working. It is necessary to erase the flash
+(with `STM32CubeProgrammer` for example) to make it work again.
```
Comment (erwango) [inline boards/arm/stm32h747i_disco/doc/index.rst:185]: Remove `` around STM32CubeProgrammer as it makes doc generation failing

**Review by erwango — CHANGES_REQUESTED** [review]

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
Failing tests: qemu_x86_64:tests/kernel/mem_protect/syscalls/kernel.memory_protection.syscalls / tests/kernel/mem_protect/syscalls/kernel.memory_protection.syscalls

**Comment by erwango** [comment]
Trying to get CI passed

**Pull request merged**